### PR TITLE
setgroups and getgroups syscalls

### PR DESCRIPTION
**Pull request by franciscozdo**
In this PR I implemented syscalls:
- `setgroups`
- `getgroups`

Added user test to check this systemcalls.

### CODE REVIEW

File: sys/kern/syscalls.c [hunk 1]
```diff
@@ -896,4 +895,53 @@ static int sys_fstatvfs(proc_t *p, fstatvfs_args_t *args, register_t *res) {
...
+  /* if we set 0 groups kmalloc returns NULL, but it is not an error */
+  if (ungroups > 0 && gidset == NULL)
+    return ENOMEM;
+
+  if ((error = copyin(ugidset, gidset, ungroups * sizeof(gid_t))))
+    goto end;
+
+  error = do_setgroups(p, ungroups, gidset);
+
+end:
```
Comment (wiklam) [inline sys/kern/syscalls.c:944]: Negating the argument allows you to remove the unnecessary goto.

**Comment by cahirwpz** [comment]
Please provide summary for this pull request.

File: sys/kern/syscalls.c [hunk 2]
```diff
@@ -896,4 +896,50 @@ static int sys_fstatvfs(proc_t *p, fstatvfs_args_t *args, register_t *res) {
...
+
+static int sys_setgroups(proc_t *p, setgroups_args_t *args, register_t *res) {
+  int error = 0;
+  int ungroups = args->ngroups;
+  const gid_t *ugidset = args->gidset;
+
+  klog("setgroups(%d, %p)", ungroups, ugidset);
+
+  /* too many groups */
+  if (ungroups > NGROUPS_MAX)
```
Comment (cahirwpz) [inline sys/kern/syscalls.c:931]: A malicious user may set `ngroups` to negative number. Have you handled that correctly?

Comment (franciscozdo) [reply sys/kern/syscalls.c]: Changed. I forgot about it.

**Pull request merged**